port/aix/omrsl.c

```diff
@@ -763,5 +763,66 @@ omrsl_startup(struct OMRPortLibrary *portLibrary)
 uintptr_t
 omrsl_get_libraries(struct OMRPortLibrary *portLibrary, OMRLibraryInfoCallback callback, void *userData)
 {
+#include <sys/ldr.h>
+#define BUFFER_SIZE 65536
+#if defined(AIXPPC)
+	uintptr_t result = 0;
+	void *buffer = NULL;
+	struct ld_info *ldInfo = NULL;
+	int rc = 0;
+	size_t bufferSize = BUFFER_SIZE;
+	for (;;) {
+		buffer = portLibrary->mem_allocate_memory(
+				portLibrary,
+				bufferSize,
+				OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+		if (NULL == buffer) {
+			portLibrary->error_set_last_error_with_message(
+					portLibrary,
+					-1,
+					"Failed to allocate memory for loadquery buffer.");
+			return -1;
+		}
+		rc = loadquery(L_GETINFO, buffer, (int)bufferSize);
+		if (-1 == rc && errno == ENOMEM) {
+			portLibrary->mem_free_memory(portLibrary, buffer);
+			bufferSize *= 2;
+		}
+		else {
+			break;
+		}
+	}
+	if (-1 == rc) {
+		int portableError = portLibrary->error_last_error_number(portLibrary);
+		portLibrary->error_set_last_error_with_message(
+				portLibrary,
+				portableError,
+				"loadquery(L_GETINFO) failed.");
+		portLibrary->mem_free_memory(portLibrary, buffer);
+		return portableError;
+	}
```

Comment:
Suggest moving the return into the loop:
```c
		rc = loadquery(L_GETINFO, buffer, bufferSize);
		if (-1 != rc) {
			break;
		}
		if (ENOMEM != errno) {
			int portableError = portLibrary->error_last_error_number(portLibrary);
			portLibrary->error_set_last_error_with_message(
					portLibrary,
					portableError,
					"loadquery(L_GETINFO) failed.");
			portLibrary->mem_free_memory(portLibrary, buffer);
			return (uintptr_t)(intptr_t)portableError;
		}
		portLibrary->mem_free_memory(portLibrary, buffer);
		bufferSize *= 2;
	}
```
Note the cast of `bufferSize` is not needed, but casts of `porableError` are required.